packages/editor/src/components/inserter-sidebar/index.js

```diff
@@ -65,6 +74,9 @@ export default function InserterSidebar() {
 						insertionPoint.insertionIndex
 					}
 					__experimentalFilterValue={ insertionPoint.filterValue }
+					__experimentalOnPatternCategorySelection={
+						handleOnCategorySelection
+					}
```

Comment:
I think we can do `__experimentalOnPatternCategorySelection={ isRightSidebarOpen ? closeGeneralSidebar : undefined }`.

This way, `onClickPatternCategory` will receive callbacks with stable reference - `closeGeneralSidebar` or `NOOP`.

Reply:
done.